### DISCUSSION TITLE
第6节课作业(2班)

### DISCUSSION THREAD

**Discussion by MING-ZCH** [InternLM/Tutorial#411] [作业]
### 基础作业
* 使用 OpenCompass 评测 InternLM2-Chat-7B 模型在 C-Eval 数据集上的性能

### 进阶作业
* 使用 OpenCompass 评测 InternLM2-Chat-7B 模型使用 LMDeploy 0.2.0 部署后在 C-Eval 数据集上的性能

**备注：由于进阶作业较难，完成基础作业之后就可以先提交作业了，在后续的大作业项目中使用这些技术将作为重要的加分点！**

**Comment by ZixinxinWang** [comment]
1 基础作业
![image](https://github.com/InternLM/tutorial/assets/152623177/ec41ffeb-3ffc-4da4-bfd5-8993db5e6253)
![image](https://github.com/InternLM/tutorial/assets/152623177/050f49a3-6c10-41a4-a16b-6386c9876b11)

**Comment by RyanWu31** [comment]
https://github.com/RyanWu31/tutorial_notebook/blob/main/README.md

**Comment by CoderJackZhu** [comment]
基础作业
使用 OpenCompass 评测 InternLM2-Chat-7B 模型在 C-Eval 数据集上的性能
![image](https://github.com/InternLM/tutorial/assets/42195616/86665b92-d2ab-4ad3-ad15-74cce9181c7d)

![image](https://github.com/InternLM/tutorial/assets/42195616/a547187f-cfed-4c9f-a184-f91c8d5138a9)
![image](https://github.com/InternLM/tutorial/assets/42195616/e6fe1e55-94e1-4f4e-9c18-7b271a92e72b)

**Comment by DarthThomas** [comment]
### 基础作业
![截屏2024-01-21 21 22 41](https://github.com/InternLM/tutorial/assets/20266717/105045cd-ca0c-48f9-860c-e7ea6e94a348)
![截屏2024-01-21 22 19 15](https://github.com/InternLM/tutorial/assets/20266717/f6950b8a-61d6-4c21-951c-6dc1ed208bf1)
![截屏2024-01-21 22 19 31](https://github.com/InternLM/tutorial/assets/20266717/32466404-ef5c-43dc-b9b2-51e58d498fce)

**Comment by dcuplover** [comment]
![6-1](https://github.com/InternLM/tutorial/assets/68799899/5e80f29a-1c8c-4e31-b0ff-8943843197b5)

**Comment by xulinui** [comment]
https://h90ag9106t.feishu.cn/docx/FpmkdNBtroX7LMxnioQcMAqJnWf

**Comment by kyrie2to11** [comment]
[InterLM L-6: OpenCompass LLM 评测](https://zhuanlan.zhihu.com/p/679226701) 打卡~

**Comment by songyuc** [comment]
### 基础作业
-   使用 OpenCompass 评测 InternLM2-Chat-7B 模型在 C-Eval 数据集上的性能


![截图 2024-01-23 00-58-26](https://github.com/InternLM/tutorial/assets/27288110/2c32e937-e817-42e4-a7ec-a6f694592bc4)

**Comment by YanxingLiu** [comment]
基础作业：
使用 OpenCompass 评测 InternLM2-Chat-7B 模型在 C-Eval 数据集上的性能
![edf65592b2e0e08e2c9393632bac9b6](https://github.com/InternLM/tutorial/assets/42299757/3c8c5186-8ccf-4339-aab0-883078a95089)
![ea4002ad1d4022411d57f2bffe4c1f7](https://github.com/InternLM/tutorial/assets/42299757/0024db01-fec9-4842-b25a-b138d182b71d)

**Comment by zyy0831** [comment]
第6节课 基础作业  使用 OpenCompass 评测 InternLM2-Chat-7B 模型在 C-Eval 数据集上的性能，只保留一条测试场景
![image](https://github.com/InternLM/tutorial/assets/43286447/2182960b-a245-43a8-ba1c-3632b2cd38f2)

**Comment by xrxu** [comment]
第6节课 基础作业 使用 OpenCompass 评测 InternLM2-Chat-7B 模型在 C-Eval 数据集上的性能

https://github.com/xrxu/Docu/blob/master/opencompass/opencompass.md

**Comment by pranerd** [comment]
**基础作业**
使用openCompass测试InternLM2-Chat-7B在C-Eval数据集上面的效果
运行命令

```
python run.py --datasets ceval_gen \
--hf-path /root/share/model_repos/internlm2-chat-7b/ \
--tokenizer-kwargs padding_side='left' truncation='left' \
trust_remote_code=True --model-kwargs trust_remote_code=True \
device_map='auto' --max-seq-len 1024 --max-out-len 16 --batch-size 2 --num-gpus 1
```
运行效果
![internlm_opencompass_2](https://github.com/InternLM/tutorial/assets/19358928/b3e3e419-eb59-4005-875a-aa6d9b9a5b36)

**进阶作业**
首先安装lmdeploy0.2.0,使用命令 `pip install lmdeploy==0.2.0`
接下来使用lmdeploy将模型进行转换
`lmdeploy convert internlm2-chat-7b /root/share/model_repos/internlm2-chat-7b --dst-path /root/model/internlm7b_lmdeploy2.0/`
#新建eval_internlm2-7b-deploy2.0.py
该脚本中的内容如下：
```
from mmengine.config import read_base
from opencompass.models.turbomind import TurboMindModel

with read_base():
    # choose a list of datasets
    from .datasets.ceval.ceval_gen_5f30c7 import ceval_datasets
    # and output the results in a choosen format
    from .summarizers.medium import summarizer


datasets = sum((v for k, v in locals().items() if k.endswith('_datasets')), [])

internlm_meta_template = dict(round=[
    dict(role='HUMAN', begin='<|User|>:', end='\n'),
    dict(role='BOT', begin='<|Bot|>:', end='<eoa>\n', generate=True),
],
                              eos_token_id=103028)

# config for internlm-chat-7b
internlm2_chat_7b = dict(
    type=TurboMindModel,
    abbr='internlm2-chat-7b-turbomind',
    path='/root/model/internlm7b_lmdeploy2.0/',
    engine_config=dict(session_len=2048,
                       max_batch_size=32,
                       rope_scaling_factor=1.0),
    gen_config=dict(top_k=1,
                    top_p=0.8,
                    temperature=1.0,
                    max_new_tokens=100),
    max_out_len=100,
    max_seq_len=2048,
    batch_size=32,
    concurrency=32,
    meta_template=internlm_meta_template,
    run_cfg=dict(num_gpus=1, num_procs=1),
)


models = [internlm2_chat_7b]
```
接下来运行命令
```
python run.py configs/eval_internlm2-7b-deploy2.0.py --debug --num-gpus 1
```
就可以出来结果了
![internlm_opencompass_3](https://github.com/InternLM/tutorial/assets/19358928/5033a9ed-e9d4-439c-91fb-286e7b67679b)

**Reply by Heitao5200** [reply]
<img width="701" alt="image" src="https://github.com/InternLM/tutorial/assets/34828079/8b9524fc-d7fb-4ce1-a6ce-f92fc1798634">
![WechatIMG1988](https://github.com/InternLM/tutorial/assets/34828079/d905a2eb-8980-4ea2-9455-817caede8c2b)
大佬 这个问题有遇到过吗？

**Comment by Mr-Vincenyang** [comment]
基础作业
![image](https://github.com/InternLM/tutorial/assets/120323418/e5530c61-799d-4073-8598-158595e3a734)

**Comment by PrinceRay7** [comment]
第六节课作业
https://github.com/PrinceRay7/InternLM_HW/blob/main/hw6/homework/homework.md

**Comment by ljqiff** [comment]
基础作业
使用 OpenCompass 评测 InternLM2-Chat-7B 模型在 C-Eval 数据集上的性能
![5](https://github.com/InternLM/tutorial/assets/29425919/101a3372-358b-4330-b68f-d557dd218b97)
![6](https://github.com/InternLM/tutorial/assets/29425919/ed9269ba-6983-4804-8034-50585dc24697)
进阶作业
使用 OpenCompass 评测 InternLM2-Chat-7B 模型使用 LMDeploy 0.2.0 部署后在 C-Eval 数据集上的性能
![6](https://github.com/InternLM/tutorial/assets/29425919/ed9269ba-6983-4804-8034-50585dc24697)

**Comment by kishiokon** [comment]
![@7JOR8)GA154T75O~H5@I A](https://github.com/InternLM/tutorial/assets/92700495/342ed1d8-ffab-4f7d-af0f-3f98fbc9b07d)
使用 OpenCompass 评测 InternLM2-Chat-7B 模型在 C-Eval 数据集上的性能

**Comment by kaoyansoft123** [comment]
基础作业：

**Reply by kaoyansoft123** [reply]
![image](https://github.com/InternLM/tutorial/assets/152302734/07656cd4-db8d-4001-9475-9cd467f6eb5b)
